### DISCUSSION TITLE
[Multiapps][3D -> 2D]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#21210] [Q&A General] [answered]
Dear Moose users,

Recently, I am trying to use both 3D and 2D geometry.

The 2D geometry is the same as a projection of 3D in the x-y plane.

I tried to transfer temperature values from the 3D to the 2D mesh.

But, it does not work well.

The temperature was not transferred from the 3D mesh to the 2D mesh at all.

I am using TransientMultiApp like below with Transfers:
---------------------------------------------------------------
```
[MultiApps]
  [sub_app] #App name that you can create in here!
    type = TransientMultiApp
    app_type = corrosionApp
    positions = '0 0 1' #x y z coordinates
    input_files = '2DTest.i'
    clone_master_mesh = false # Different mesh
    execute_on = 'TIMESTEP_END'
    catch_up = true
  []
[]

[Transfers]
  [tosub]
    type = MultiAppProjectionTransfer
    execute_on = 'TIMESTEP_END'
    to_multi_app = sub_app
    source_variable = 'T'
    variable = 'T'
  []
[]
```
---------------------------------------------------------------

Are there any mistakes in here?

Best regards

**Comment by GiudGiud** [comment, marked as answer]
Hello

The projection transfer is not a geometrical projection across dimensions, it s a projection between different meshes and finite element basis that still share the same geometrical space. 

From the 3D to the 2D, there is not a unique mapping. Are you trying to average the 3D temperatures axially? Or pick a slice in the 3D system? 

Guillaume

**Reply by coskrrb2002** [reply]
Thank you for your answer,


I am trying to pick a slice in the 3D system (surface) and moved the calculated data to the 2D model.

Is there any other way to do this?


Nakkyu

**Reply by GiudGiud** [reply]
do the two meshes overlap?
What is the extent of the 2D mesh? of the 3D mesh?

**Reply by coskrrb2002** [reply]
I would like to do the following things (transient model - time dependent):
Calculate temperature profile in a 3D mesh. (A box shape)
Transfer temperature into a 2D mesh. (One of the plane on the box)
Calculate the temperature dependent physics in the 2D mesh.

I know if I apply the temperature dependent physics only on the surface, then I don't have to calculate seperately (3D and 2D).

However, in my case, I have to use different coordinate systems. 2D mesh: cylindrical coordinate and 3D mesh: cartesian coordiante.

That's why I want to seperate into 3D and 2D mesh.

If there any other methods in MOOSE, probably I can extract temperature profile from 3D mesh and apply to 2D mesh as boundary conditions...

**Reply by GiudGiud** [reply]
hello

Since you have a coordinate transfer change in there as well, you will want to use this PR:
https://github.com/idaholab/moose/pull/20451
It also provides additional documentation on dimensional transfers. I m not sure 3D-2D RZ is allowed currently

Guillaume

**Reply by coskrrb2002** [reply]
Thank you for your advice!
